Handle channel IDs without index or modifier correctly

hshmt opened the pull request:
Channel whose ID is "timestamp" has type IIO_CHAN_TYPE_UNKNOWN
before this patch, but it should be IIO_TIMEOUT.

rgetz commented:
@hshmt : what driver is this?

When I look at devices that I'm familiar with:
https://wiki.analog.com/resources/tools-software/linux-drivers/iio-inertial-measurement-units/adis16475#hardware_configuration

```
-rw-r--r-- 1 root root 4096 Jul 12 10:17 current_timestamp_clock
-rw-r--r-- 1 root root 4096 Jul 12 10:17 in_timestamp_en
-r--r--r-- 1 root root 4096 Jul 12 10:17 in_timestamp_index
-r--r--r-- 1 root root 4096 Jul 12 10:17 in_timestamp_type
```

-Robin

hshmt commented:
@rgetz I'm working on [Chromium OS EC](https://chromium.googlesource.com/chromiumos/platform/ec/+/master/README.md), and I'm seeing the same timestamp-related entries under scan_elements:
```
-rw-r--r--. 1 root root 4096 Feb  6 18:39 in_timestamp_en
-r--r--r--. 1 root root 4096 Feb  6 18:39 in_timestamp_index
-r--r--r--. 1 root root 4096 Feb  6 18:39 in_timestamp_type
```

iio_channel_get_id() for this timestamp channel returns "timestamp", and currently its type is set to IIO_CHAN_TYPE_UNKNOWN.
I guess the type was (probably unintentionally) set to IIO_TIMESTAMP correctly before 25c20d67addf55a22e3e2a9c0ba0abcae5733fb5 because `chn->id[len] != '_' && chn->id[len] < '0' && chn->id[len] > '9'` is always false.

rgetz commented:
Thanks - I don't have the Chrome EC, but @nunojsa has one of the IMUs to try it on.

Thanks for confirming.

-Robin

nunojsa commented:
@hshmt, I have IMUs where I can try this. Nevertheless, I think you need to add a `Signed-off-by:` line in your patch. Also, on the commit message, don't you mean "...but it should be __IIO_TIMESTAMP__."?

hshmt commented:
> @hshmt, I have IMUs where I can try this. Nevertheless, I think you need to add a `Signed-off-by:` line in your patch. Also, on the commit message, don't you mean "...but it should be **IIO_TIMESTAMP**."?

My bad. Added a Signed-off-by line and fixed the typo to IIO_TIMESTAMP.

nunojsa commented:
No worries. I will try to have feedback on this latest tomorrow...

hshmt commented:
Thank you for reviewing!
Rebased onto master.
